### PR TITLE
Add hardware_threads() function

### PR DESCRIPTION
**Pull request by IUdalov**
Fixes https://github.com/effil/effil/issues/101

### CODE REVIEW

File: tests/lua/thread.lua [hunk 1]
```diff
@@ -4,6 +4,10 @@ local effil = effil
 
 test.thread.tear_down = default_tear_down
 
+test.thread.hardware_threads = function()
+    test.is_number(effil.hardware_threads())
```
Comment (mihacooper) [inline tests/lua/thread.lua]: It should be >= 0, as well

File: README.md [hunk 2]
```diff
@@ -304,6 +305,10 @@ Explicit cancellation point. Function checks *cancellation* or *pausing* flags o
 ### `effil.sleep(time, metric)`
 Suspend current thread.
 
+### `effil.hardware_threads()`
+Returns the number of concurrent threads supported by implementation.
+Basically forwards value from [std::thread::hardware_concurrency](https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency).
+
 **input**: [time metrics](#time-metrics) arguments.
```
Comment (mihacooper) [inline README.md]: This line refer to effil.sleep, wrong place to insert

Comment (IUdalov) [reply README.md]: Fixed.

File: tests/lua/thread.lua (hunk 1 above)
Comment (IUdalov) [reply tests/lua/thread.lua]: done.

**Pull request merged**